dcp-prototype/frontend/browser/openapi.yml

```diff
@@ -0,0 +1,106 @@
+openapi: 3.0.0
+info:
+  description: |
+    DCP 2.0 Consumer backend API
+  version: "1.0.0"
+  title: DCP 2.0 Consumer 
+servers:
+# Added by API Auto Mocking Plugin
+  - description: SwaggerHub API Auto Mocking
+    url: 'https://api.browser.humancellatlas.org/v1'
+tags:
+  - name: project
+    description: API used to access project information
+  - name: download
+    description: API used to download data
+paths:
+  /projects:
+    get:
+      tags:
+        - project
+      summary: Retrieve filterable information on all projects
+      operationId: getProjects
+      responses:
+        '200':
+          description: OK
+  /projects/{projectUUID}:
+    get:
+      tags:
+        - project
+      summary: |
+        Retrieve detailed information about a project. This includes a list of 
+        downloadable files.
+      operationId: getProjectsUuid
+      parameters: 
+        - name: projectUUID
+          in: path
+          description: The uuid of the project
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: OK
+        '400':
+          description: Bad Request
+        '404':
+          description: File Not Found
+  /projects/{projectUUID}/files/{fileUUID}:
```

Comment:
What are your thoughts on revising this to `/projects/<project_uuid>/files` for which returns a list of files, their metadata and S3 identifier?

The advantage of this would be to reduce the number of API calls to render the [Project page and files](https://www.figma.com/proto/LCBTk5cTkoOtW9IC5GfpeF/2-Browser?node-id=49%3A3&scaling=min-zoom) (2 calls in the proposed design, and `num_files` calls for the current design). This assumes the number of files in a project is manageable.

Reply:
My assumption was that the files in the project would be included in the `/projects/<project_uuid>` endpoint

Reply:
I mainly question here the value of an endpoint that returns a single field about a file i.e., `s3_location`. What I'm proposing alternatively is `s3_location` is returned along with all of the other file metadata fields. @colinmegill as our client, do you have an opinion here?

Reply:
I think it will have to get a signed URL for every file, so you might not want to do that for all files in a project if you don't need to.

Reply:
Though I'm also not sure what you get out of nesting `files` under project here. Since you are identifying files by their own uuid, isn't the project uuid unneeded? Seems like you could move it to just `/files/{file_uuid}`

Reply:
@mckinsel it's nested under projects since the unit of access control should be a project.

Reply:
@Bento007 This detail can be handled on the backend (e.g. `project_uuid` will be a field in the `files` table) and I would opt for keeping the endpoint as minimal as possible

Reply:
> @mckinsel it's nested under projects since the unit of access control should be a project.

Could you say more about this @Bento007?

Reply:
@calvinnhieu so a file id will be unique within a project but not unique relative to all other files? According to @ambrosejcarr the same file can be in multiple projects. I guess it comes down to how we decided to store and group the data and what level of access control we want in the future.

Reply:
> According to @ambrosejcarr the same file can be in multiple projects.

This is a somewhat distant future that I don't believe we need to design towards right now. For the time being I would assume each file is associated with one project. Could you point me at the place I wrote this down? I'd like to correct the language to make it clearer.

The use case is that a tissue network such as the kidney submits a series of projects. At this point, all files are associated with only one project. Then, they go do a meta analysis and that combines the count matrices from all experiments into one large count matrix. The count matrix is a new file, associated only with the meta-analysis. However, we may want to link the raw fastq and bam files that were used in the meta-analysis, which would mean each of those files is now linked to two experiments. There are alternatives though, which might be to simply nest the projects. 

>> @mckinsel it's nested under projects since the unit of access control should be a project.
>
> Could you say more about this @Bento007?

My question above was with respect to "unit of access control".

Reply:
@calvinnhieu according to @ambrosejcarr a file can be in multiple projects. How would you differentiate which files are in which project? Maybe an additional table of file to projects. If you're ok with handling it in the backend then sure.

Reply:
@ambrosejcarr as far as I see it we have three main tiers of access granularity we can go with, at the artifact, project, or file level. Projects I see as being a logical unit of access since it is a natural grouping of files and is the thing uses will want to see and share. Putting the files under projects is my way of suggesting that this should be how we manage access control of the files.

Reply:
You'd need map files to project (or whatever the access control unit is) in the backend anyway, right? You couldn't just rely on the project uuid in the path.

Reply:
If this is a hard blocker then I will change it. It can always be improved later.